.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,66 @@
+# Run performance benchmarks
+#
+# Continuous benchmarking using pytest-codspeed. Measures the execution speed
+# of tests marked with @pytest.mark.benchmark decorator.
+
+name: Benchmarks
+
+on:
+  # Run on pushes to the main branch
+  push:
+    branches: [ main ]
+  # Uncomment the 'pull_request' line below to trigger the workflow in PR
+  pull_request:
+  # `workflow_dispatch` allows CodSpeed to trigger backtest
+  # performance analysis in order to generate initial data.
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      # Checkout current git repository
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          # fetch all history so that setuptools-scm works
+          fetch-depth: 0
+
+      # Install Miniconda with conda-forge dependencies
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3.0.1
+        with:
+          auto-activate-base: true
+          activate-environment: ""  # base environment
+          channels: conda-forge,nodefaults
+          channel-priority: strict
+
+      # Install GMT and dependencies from conda-forge
+      - name: Install dependencies
+        run: |
+          conda install --solver=libmamba gmt=6.4.0 python=3.12 \
+                        numpy pandas xarray netCDF4 packaging \
+                        pytest pytest-benchmark pytest-mpl
+          $CONDA/bin/python -m pip install -U pytest-codspeed setuptools
+
+      # Install the package that we want to test
+      - name: Install the package
+        run: $CONDA/bin/python -m pip install --no-deps -e .
+
+      # Run the benchmark tests
+      - name: Run benchmarks
+        uses: CodSpeedHQ/action@v2
+        with:
+          run: |
+            $CONDA/bin/python -c "import pygmt; pygmt.show_versions()"
+            PYGMT_USE_EXTERNAL_DISPLAY="false" $CONDA/bin/python -m pytest -r P --pyargs pygmt --codspeed
```

Comment:
Hmm, the benchmark tests seems to be running fine on CI, but the upload step doesn't work. E.g. at https://github.com/GenericMappingTools/pygmt/actions/runs/7312581712/job/19923300422?pr=2908#step:6:121:

```python-traceback
   ============================= test session starts ==============================
  platform linux -- Python 3.12.1, pytest-7.4.3, pluggy-1.3.0 -- /usr/share/miniconda/bin/python
  codspeed: 2.2.0 (callgraph: enabled)
  CodSpeed had to disable the following plugins: pytest-benchmark
  cachedir: .pytest_cache
  benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
  Matplotlib: 3.8.2
  Freetype: 2.12.1
  rootdir: /home/runner/work/pygmt/pygmt
  configfile: pyproject.toml
  plugins: benchmark-4.0.0, codspeed-2.2.0, mpl-0.16.1
  collecting ... collected 679 items / 677 deselected / 8 skipped / 2 selected
  
  pygmt/tests/test_blockm.py::test_blockmean_input_dataframe PASSED        [ 50%]
  pygmt/tests/test_grd2xyz.py::test_grd2xyz PASSED                         [100%]
  ================================ 2 benchmarked =================================
  
  =============================== warnings summary ===============================
  ../../../../../usr/share/miniconda/lib/python3.12/site-packages/dateutil/tz/tz.py:37
    /usr/share/miniconda/lib/python3.12/site-packages/dateutil/tz/tz.py:37: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
      EPOCH = datetime.datetime.utcfromtimestamp(0)
  
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  ==================================== PASSES ====================================
  ============================== slowest durations ===============================
  4.42s call     pygmt/tests/test_blockm.py::test_blockmean_input_dataframe
  3.04s setup    pygmt/tests/test_blockm.py::test_blockmean_input_dataframe
  1.85s call     pygmt/tests/test_grd2xyz.py::test_grd2xyz
  1.16s setup    pygmt/tests/test_grd2xyz.py::test_grd2xyz
  
  (2 durations < 0.2s hidden.  Use -vv to show these durations.)
  =========== 2 passed, 8 skipped, 677 deselected, 1 warning in 21.84s ===========
Upload the results
  CodSpeed Run Hash: "8332511d76e690d4a1ef10b5df14108240ab0ccb8be79e5059092401c7a5ec31"
  Preparing upload...
  Error: error decoding response body: missing field `status` at line 1 column 58
  Error: Process completed with exit code 1.
```

Reply:
Ah, it might be because we need to install it first (https://docs.codspeed.io/ci/github-actions#1-allow-the-action-to-upload-on-codspeed). @seisman, could you approve the installation request please?

![image](https://github.com/GenericMappingTools/pygmt/assets/23487320/3575744f-5270-4711-af61-aa0f70743940)

Reply:
Approved.

Reply:
Hmm, I don't think I have admin permissions to enable the PyGMT repo on https://codspeed.io/settings/organizations/GenericMappingTools/general. Could you do it on your side @seisman?

![Screenshot_2023-12-24-22-15-34-56_3aea4af51f236e4932235fdada7d1643.jpg](https://github.com/GenericMappingTools/pygmt/assets/23487320/07fc64a6-06bd-4320-89c0-c4115a158882)

Reply:
Done. FYI, I also added the `CODSPEED_TOKEN` token following the instructions.

Reply:
Thanks! Let me try re-running the tests first without the token.